ner_v2/detectors/numeral/number_range/README.md

```diff
@@ -0,0 +1,246 @@
+   | se niche \| se kam \| se sasta \| se saste k aas paas\|  k aas pas\| k lagbhag \|  से नीचे \| से कम \| से सस्ता \| से सस्ते \|   के आस पास  \| के लगभग |    1     |    max     |
+   |                           se\|-\|से                           |    0     |  min_max   |
+
+   Here, the 1st column will contain the variants of range keywords separated by pipes, which are present before or after or in between number values and defines what type of number value is present, either min or max. 
+
+   2rd column corresponds to the position of the keyword. If 1 means it will be present after number value, -1 if present before number.  
+
+   3rd column defines the range type of number i.e if it contains min value or max value or both.    
+
+   **For example** - *mujhe 2000 se jada log chahiye kal ki rally me* . In this text, range keyword `se jada` which is present in position `1` (after number) defines that number `2000` is the minimum value defined in the text, since no maximum value specified in text, hence it will be null.
+
+
+#### Guidelines to add new detectors for number range apart from builtin ones:
+
+Create a new class `number_range_detection.py`  inside language directory.
+To start with copy the code below
+
+```python
+import re
+import os
+
+from ner_v2.constant import LANGUAGE_DATA_DIRECTORY
+from ner_v2.detectors.numeral.constant import NUMBER_REPLACE_TEXT
+from ner_v2.detectors.numeral.number_range.standard_number_range_detector import BaseNumberRangeDetector
+
+
+class NumberRangeDetector(BaseNumberRangeDetector):
+    data_directory_path = os.path.join((os.path.dirname(os.path.abspath(__file__)).rstrip(os.sep)),
+                                       LANGUAGE_DATA_DIRECTORY)
+
+    def __init__(self, entity_name, language, unit_type=None):
+        super(NumberRangeDetector, self).__init__(entity_name=entity_name,
+                                                  language=language,
+                                                  data_directory_path=NumberRangeDetector.data_directory_path,
+                                                  unit_type=unit_type)
+```
+
+Note that the class name must be `NumberRangeDetector` 
+and should inherit from `ner_v2.detectors.numeral.number_range.standard_number_range_detector.BaseNumberRangeDetector`
+
+Next, we define a custom detector. For our purposes, we will add a detector to detect number range from text of patter 'between *number1* and *number2*. It will detect number1 and number2 as minimum value and maximum value respectively.
+
+1. The custom detector must accept two arguments `number_range_list` and `original_list` and must operate on `self.processed_text (This text contained tagged number text i.e number will be replaced with __dnumber__<number>`). So in your pattern, you just have to include pattern with number tag instead of `\d+` for the number. (as it handle all number having decimal, ordinal and integer)
```

Comment:
```
You must understand that self.processed_text is not the text passed into the `detect` or `detect_entity` function.
Internally BaseNumberRangeDetector runs NumberDetector on the passed in text to tag it with numbers as this architecture allows us to write simple patterns without worrying about how numbers actually appear (decimal, ordinal, integers, etc) in the text

E.g. `more than 3 but less than 2.5` -> `more than __number__1 but less than __number__0`

Your custom patterns must operate on the latter, so instead of writing a `\d+` for numbers, you must write `'{number}\d+'.format(ner_v2.detectors.numeral.constant.NUMBER_REPLACE_TEXT)`. Please refer to the code block below to see an example of writing custom pattern